tests/suite/system_suite_test.go

```diff
@@ -101,38 +130,84 @@ var _ = BeforeSuite(func() {
 		version = "edge"
 	}
 
-	clusterInfo, err = resourceManager.GetClusterInfo()
-	Expect(err).ToNot(HaveOccurred())
-
 	output, err := framework.InstallGatewayAPI(k8sClient, *gatewayAPIVersion, *k8sVersion)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
-	output, err = framework.InstallNGF(cfg)
+	output, err = framework.InstallNGF(installCfg, extraInstallArgs...)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
-	podName, err := framework.GetNGFPodName(k8sClient, cfg.Namespace, cfg.ReleaseName, timeoutConfig.CreateTimeout)
+	podNames, err := framework.GetNGFPodNames(
+		k8sClient,
+		installCfg.Namespace,
+		installCfg.ReleaseName,
+		timeoutConfig.CreateTimeout,
+	)
 	Expect(err).ToNot(HaveOccurred())
 
 	if *serviceType != "LoadBalancer" {
-		portFwdPort, err = framework.PortForward(k8sConfig, cfg.Namespace, podName, portForwardStopCh)
+		portFwdPort, err = framework.PortForward(k8sConfig, installCfg.Namespace, podNames[0], portForwardStopCh)
 		address = "127.0.0.1"
 	} else {
-		address, err = resourceManager.GetLBIPAddress(cfg.Namespace)
+		address, err = resourceManager.GetLBIPAddress(installCfg.Namespace)
 	}
 	Expect(err).ToNot(HaveOccurred())
-})
+}
 
-var _ = AfterSuite(func() {
-	portForwardStopCh <- struct{}{}
+func teardown() {
+	if portFwdPort != 0 {
+		portForwardStopCh <- struct{}{}
+	}
 
 	cfg := framework.InstallationConfig{
-		ReleaseName: "ngf-test",
-		Namespace:   "nginx-gateway",
+		ReleaseName: releaseName,
+		Namespace:   ngfNamespace,
 	}
 
 	output, err := framework.UninstallNGF(cfg, k8sClient)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
 	output, err = framework.UninstallGatewayAPI(*gatewayAPIVersion, *k8sVersion)
 	Expect(err).ToNot(HaveOccurred(), string(output))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	Expect(wait.PollUntilContextCancel(
```

Comment:
Can we add the `--wait` arg to the helm uninstall command instead of using this wait?

Reply:
helm doesn't delete the namespace, which is why this is here